src/main/java/seedu/address/logic/parser/FindNumCommandParser.java

```diff
@@ -0,0 +1,34 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.Arrays;
+
+import seedu.address.logic.commands.FindNumCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.PhoneContainsDigitsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindNumCommand object
+ */
+public class FindNumCommandParser implements Parser<FindNumCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindNumCommand
+     * and returns a FindNumCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindNumCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindNumCommand.MESSAGE_USAGE));
+        }
+
+        String[] phoneKeywords = trimmedArgs.split("\\s+");
+
+        return new FindNumCommand(new PhoneContainsDigitsPredicate(Arrays.asList(phoneKeywords)));
+    }
```

Comment:
could do phone number validation in here to check for only numerical values given. see Phone#isValidPhone